syntax = "proto3";
import "common.proto";
import "backup.proto";
package huawei.proto.inner;

service BackupInnerControl {
    rpc Create(CreateBackupInReq) returns(CreateBackupInAck){}
    rpc Delete(DeleteBackupInReq) returns(DeleteBackupInAck){}
    rpc Restore(RestoreBackupInReq) returns(stream RestoreBackupInAck){}
    rpc Get(GetBackupInReq) returns(GetBackupInAck){}
    rpc List(ListBackupInReq) returns(ListBackupInAck){}
}

message CreateBackupInReq {
    string vol_name = 1; 
    uint64 vol_size = 2;
    string backup_name = 3;
    BackupOption backup_option = 4;
}

message CreateBackupInAck {
    StatusCode status = 1;
}

message DeleteBackupInReq {
    string vol_name = 1; 
    string backup_name = 2;
}

message DeleteBackupInAck {
    StatusCode status = 1;
}

message RestoreBackupInReq {
    string vol_name = 1; 
    string backup_name = 2;
}

message RestoreBackupInAck {
    StatusCode status = 1;
    uint64 blk_no     = 2;
    /*local site restore return block store object name*/
    string blk_obj    = 3;
    /*remote site restore return block store object data*/
    bytes  blk_data   = 4;
    /*no any blk any more*/
    bool   blk_over = 5;
}

message GetBackupInReq {
    string vol_name = 1; 
    string backup_name = 2;
}

message GetBackupInAck {
    StatusCode status = 1;
    BackupStatus backup_status = 2;
}

message ListBackupInReq {
    string vol_name = 1; 
}

message ListBackupInAck {
    StatusCode status = 1;
    repeated string backup_name = 2;
}
